### ISSUE TITLE
Producer.flow cannot transmit an offset without a message

### ISSUE THREAD

**Issue by patrick-premont** [akka/reactive-kafka#324]
When consuming and producing to Kafka, we usually want to commit the `commitableOffset` after the corresponding messages have been produced. When there are no messages to produce, possibly because the consumed message was erroneous, we would still like to commit the source offset.

`Producer.flow` offers the `PassThrough` type parameter to transmit extra information, such as source offsets, but as explained above, we sometimes we need to pass information trough in the absence of messages to be produced.

One way to achieve this would be to add a more general method `Produce.eitherFlow[K, V, PassThrough, E]` which would create a
```
Flow[Either[E, Message[K, V, PassThrough]], Either[E, Result[K, V, PassThrough]], NotUsed]
``` 

In the use-case discussed here, both the `E` and `PassThrough` type parameters would be `CommitableOffset`.

The current workarounds are either to omit to commit some offsets or to drop down to the Kafka API to produce messages, as illustrated here for example: https://www.iravid.com/posts/using-cats-with-reactive-kafka.html.

**Comment by DWvanGeest** [comment]
This is related to #415 it seems. I would love to see this fixed.

**Comment by DWvanGeest** [comment]
This is solved by #495.

**Comment by ennru** [comment]
Thank you for pointing to this, yes it is now supported.
